### PR TITLE
INT-1058

### PR DESCRIPTION
**Pull request by pgarcess**

### CODE REVIEW

File: controllers/paymentController.php [hunk 1]
```diff
@@ -360,7 +363,6 @@ public function pagantisReceiptPage($order_id)
             $pagantisOrder = $orderClient->createOrder($orderApiClient);
             if ($pagantisOrder instanceof \Pagantis\OrdersApiClient\Model\Order) {
                 $url = $pagantisOrder->getActionUrls()->getForm();
-                addOrderToCartProcessingQueue($order->get_id(), $pagantisOrder->getId());
```
Comment (rmarinleal) [inline controllers/paymentController.php:363]: it's no longer necessary?

File: controllers/paymentController4x.php [hunk 2]
```diff
@@ -326,7 +326,6 @@ public function pagantisReceiptPage($order_id)
             $pagantisOrder = $orderClient->createOrder($orderApiClient);
             if ($pagantisOrder instanceof \Pagantis\OrdersApiClient\Model\Order) {
                 $url = $pagantisOrder->getActionUrls()->getForm();
-                addOrderToCartProcessingQueue($order->get_id(), $pagantisOrder->getId());
```
Comment (rmarinleal) [inline controllers/paymentController4x.php:329]: it's no longer necessary?

File: includes/pg-functions.php [hunk 3]
```diff
@@ -158,13 +161,8 @@ function updateThousandsSeparatorDbConfig()
...
     $tableName         = $wpdb->prefix . PG_CONFIG_TABLE_NAME;
     $thousandSeparator = get_option('woocommerce_price_thousand_sep');
-    $wpdb->update(
-        $tableName,
-        array('value' => $thousandSeparator),
-        array('config' => 'PAGANTIS_SIMULATOR_THOUSANDS_SEPARATOR'),
-        array('%s'),
-        array('%s')
-    );
+    $wpdb->update($tableName, array('value' => $thousandSeparator), array('config' => 'PAGANTIS_SIMULATOR_THOUSANDS_SEPARATOR'),
```
Comment (rmarinleal) [inline includes/pg-functions.php]: Once you use multiline to call a funciton, each parameter have to be in it's own line. 
https://www.php-fig.org/psr/psr-2/#44-method-arguments

File: includes/pg-functions.php [hunk 4]
```diff
@@ -175,13 +173,8 @@ function updateDecimalSeparatorDbConfig()
...
     $tableName        = $wpdb->prefix . PG_CONFIG_TABLE_NAME;
     $decimalSeparator = get_option('woocommerce_price_decimal_sep');
-    $wpdb->update(
-        $tableName,
-        array('value' => $decimalSeparator),
-        array('config' => 'PAGANTIS_SIMULATOR_DECIMAL_SEPARATOR'),
-        array('%s'),
-        array('%s')
-    );
+    $wpdb->update($tableName, array('value' => $decimalSeparator), array('config' => 'PAGANTIS_SIMULATOR_DECIMAL_SEPARATOR'),
```
Comment (rmarinleal) [inline includes/pg-functions.php]: https://www.php-fig.org/psr/psr-2/#44-method-arguments

File: includes/pg-functions.php [hunk 5]
```diff
@@ -255,21 +248,22 @@ function isPluginEnabled()
...
     $settings = get_option('woocommerce_pagantis_settings');
-    return (! empty($settings['enabled_4x']) && 'yes' === $settings['enabled_4x']);
+
+    return ( ! empty($settings['enabled_4x']) && 'yes' === $settings['enabled_4x']);
 }
 
 
 function isCountryShopContextValid()
 {
     $locale           = strtolower(strstr(get_locale(), '_', true));
```
Comment (rmarinleal) [inline includes/pg-functions.php]: Delete multiple spaces before equal "="

File: WC_Pagantis.php [hunk 6]
```diff
@@ -31,7 +29,8 @@
 define('PG_CONFIG_TABLE_NAME', 'pagantis_config');
 define('PG_LOGS_TABLE_NAME', 'pagantis_logs');
 define('PG_CONCURRENCY_TABLE_NAME', 'pagantis_concurrency');
-define('PG_CART_PROCESS_TABLE', 'cart_process');
+define('PG_OLD_CART_PROCESS_TABLE', 'cart_process');
+define('PG_ORDER_PROCESS_TABLE', 'pagantis_order');
```
Comment (pgarcess) [inline WC_Pagantis.php:33]: If you want to create a new table with "better" name, please rename the table initially, then you will to maintain only one table. Use this mysql command: https://www.techonthenet.com/mysql/tables/alter_table.php#:~:text=The%20syntax%20to%20rename%20a%20table%20in%20MySQL%20is%3A,The%20table%20to%20rename.

File: WC_Pagantis.php [hunk 7]
```diff
@@ -255,11 +254,17 @@ public function pagantisActivation()
             }
         }
 
+        // Creating new cart processing table < 8.6.13
+        if (isPgTableCreated(PG_OLD_CART_PROCESS_TABLE)) {
+            alterCartProcessingTable();
```
Comment (pgarcess) [inline WC_Pagantis.php:259]: In this function you need to rename the table too.

File: controllers/notifyController.php [hunk 8]
```diff
@@ -109,23 +115,17 @@ public function processInformation()
...
-        if ($this->woocommerceOrderId == '') {
-            throw new QuoteNotFoundException();
-        }
-
         $this->unblockConcurrency();
         $this->blockConcurrency($this->woocommerceOrderId);
     }
 
     /**
      * getProductType
```
Comment (pgarcess) [inline controllers/notifyController.php]: Wrong comment

File: config/docker-setup.sh [hunk 9]
```diff
@@ -0,0 +1 @@
+#!/bin/bash
```
Comment (pgarcess) [inline config/docker-setup.sh:1]: Remove this file. New file with empty content and we're not using it

File: controllers/notifyController.php [hunk 10]
```diff
@@ -54,24 +53,31 @@ class WcPagantisNotify extends WcPagantisGateway
...
+
+                $this->setWoocommerceOrderId();
+                $this->setVerificationToken();
+                $this->verifyOrderConformity();
                 $this->checkConcurrency();
-                $this->getProductType();
+                $this->setPaymentProductType();
                 $this->getMerchantOrder();
-                $this->getPagantisOrderId();
+                $this->setPagantisOrderId();
```
Comment (pgarcess) [inline controllers/notifyController.php]: Please keep the conventional naming: https://digitalorigin.atlassian.net/wiki/spaces/MERCHD/pages/599752794/Order+Confirmation+Flow

File: controllers/notifyController.php [hunk 11]
```diff
@@ -148,17 +148,50 @@ private function getMerchantOrder()
     }
 
     /**
-     * @throws NoIdentificationException
+     * @throws MerchantOrderNotFoundException
      */
-    private function getPagantisOrderId()
+    private function verifyOrderConformity()
```
Comment (pgarcess) [inline controllers/notifyController.php:152]: It is done inside setPagantisOrderId function

File: controllers/notifyController.php [hunk 12]
```diff
@@ -226,8 +259,9 @@ private function checkMerchantOrderStatus()
                           " Merchant order id:".$this->woocommerceOrder->get_id().
                           " Merchant order status:".$this->woocommerceOrder->get_status().
                           " Pagantis order id:".$this->pagantisOrder->getStatus().
-                          " Pagantis order status:".$this->pagantisOrder->getId();
-
+                          " Pagantis order status:".$this->pagantisOrder->getId().
+                          " Pagantis urlToken: ".$this->getVerificationToken();
```
Comment (pgarcess) [inline controllers/notifyController.php]: It is not a revelant info for the mechant

File: controllers/notifyController.php [hunk 13]
```diff
@@ -395,14 +425,14 @@ private function updateBdInfo()
         global $wpdb;
 
         $this->checkDbTable();
-        $tableName = $wpdb->prefix.PG_CART_PROCESS_TABLE;
+        $tableName = $wpdb->prefix . PG_OLD_CART_PROCESS_TABLE;
```
Comment (pgarcess) [inline controllers/notifyController.php]: If it's a new installation the tables won't exist.

File: controllers/paymentController.php [hunk 14]
```diff
@@ -360,7 +367,13 @@ public function pagantisReceiptPage($order_id)
...
+                addOrderToProcessingQueue(WC()->cart->get_cart_hash(), $pagantisOrder->getId(), $order->get_id(), $this->paymentProcessingToken);
```
Comment (pgarcess) [inline controllers/paymentController.php]: Please keep the command $order->get_id()

File: controllers/paymentController.php [hunk 15]
```diff
@@ -55,6 +60,7 @@ public function __construct()
         $this->id = WcPagantisGateway::METHOD_ID;
         $this->has_fields = true;
         $this->method_title = ucfirst($this->id);
+        $this->paymentProcessingToken = strtoupper(md5(uniqid(rand(), true)));
```
Comment (pgarcess) [inline controllers/paymentController.php]: Please change it, we told you twice:  
- https://pagantis.slack.com/archives/G017X8BB0C8/p1600860303011700
- https://pagantis.slack.com/archives/G017X8BB0C8/p1600854426008600

File: controllers/paymentController.php [hunk 16]
```diff
@@ -45,6 +45,11 @@ class WcPagantisGateway extends WC_Payment_Gateway
      */
     private $allowed_currencies;
 
+    /**
+     * @var string
+     */
+    private $paymentProcessingToken;
```
Comment (pgarcess) [inline controllers/paymentController.php]: Please change it, we told you twice:  
- https://pagantis.slack.com/archives/G017X8BB0C8/p1600860303011700
- https://pagantis.slack.com/archives/G017X8BB0C8/p1600854426008600

File: controllers/paymentController.php [hunk 17]
```diff
@@ -306,7 +312,8 @@ public function pagantisReceiptPage($order_id)
             $callback_arg = array('wc-api'=>'wcpagantisgateway',
                 'key'=>$order->get_order_key(),
                 'order-received'=>$order->get_id(),
-                'origin' => ''
+                'origin' => '',
+                'token' => $this->paymentProcessingToken
```
Comment (pgarcess) [inline controllers/paymentController.php]: 'token' => $this->urlToken

File: controllers/paymentController.php [hunk 18]
```diff
@@ -620,7 +634,7 @@ private function getKoUrl($order)
     private function getKeysUrl($order, $url)
     {
         $defaultFields = (get_class($order)=='WC_Order') ?
-            array('order-received'=>$order->get_id(), 'key'=>$order->get_order_key()) :
+            array('order-received'=>$order->get_id(), 'key'=>$order->get_order_key(), 'token' => $this->paymentProcessingToken) :
```
Comment (pgarcess) [inline controllers/paymentController.php]: Remove it, it is only for internal redirection

File: controllers/paymentController4x.php [hunk 19]
```diff
@@ -48,16 +48,24 @@ class WcPagantis4xGateway extends WC_Payment_Gateway
      */
     private $allowed_currencies;
 
+    /**
+     * @var string
+     */
+    private $paymentProcessingToken4x;
```
Comment (pgarcess) [inline controllers/paymentController4x.php]: Same changes that in paymentController.php...

File: controllers/paymentController.php (hunk 15 above)
Comment (tmeliotpg) [reply controllers/paymentController.php]: if the name is identical in both payment methods it create issues so I have renamed it to ` $urlToken` for the normal method and ` $urlToken4x` for the 4x method

File: controllers/paymentController.php (hunk 16 above)
Comment (tmeliotpg) [reply controllers/paymentController.php]: if the name is identical in both payment methods it create issues so I have renamed it to ` $urlToken` for the normal method and ` $urlToken4x` for the 4x method

File: controllers/notifyController.php (hunk 11 above)
Comment (tmeliotpg) [reply controllers/notifyController.php:152]: moved it to getPagantisOrderId

File: controllers/notifyController.php (hunk 10 above)
Comment (tmeliotpg) [reply controllers/notifyController.php]: done

File: controllers/notifyController.php (hunk 12 above)
Comment (tmeliotpg) [reply controllers/notifyController.php]: modified

File: controllers/notifyController.php (hunk 13 above)
Comment (tmeliotpg) [reply controllers/notifyController.php]: fixed

**Pull request closed without merging**